### PR TITLE
Add first version of script computing the aggregated stats for a release

### PR DESCRIPTION
**Pull request by sbesson**
Consumed by https://github.com/IDR/idr.openmicroscopy.org/pull/94 /cc @pwalczysko

### CODE REVIEW

**Comment by pwalczysko** [comment]
The script seems to be adding newlines after every run, but not taking these into account when the script is run again on the same file. This results in ampty lines.

```
...
2020-01-16	prod73	0.7.3	12544	1188147		8862811	62872383	158.30338343134298	21.780547	423

2020-03-03	prod80	0.8.0	12569	1195544		9083954	63756762	158.91306092503	22.181109	431
```

**Comment by pwalczysko** [comment]
The column of Experiments? seems to being missed by this script, is that intentional ?

```
2020-06-30	prod84	0.8.4	12935	1210586	1178081	9104040	64886054	180.8749	23.23	358
2020-07-22	prod85	0.8.5	13027	1210586	1178173	9106609	65211327	181.1312	23.24	359 
2020-01-16	prod73	0.7.3	12544	1188147		8862811	62872383	158.30338343134298	21.780547	423

2020-03-03	prod80	0.8.0	12569	1195544		9083954	63756762	158.91306092503	22.181109	431
```

The first two rows above are from the previous releases.tsv, the two new lines were added by this script.

**Comment by sbesson** [comment]
Yes we are now omitting `Experiments` and `Targets` as these concepts need to be redefined (but the schema of the spreadsheet is still used elsewhere). I will look into the empty rows while generating the stats for `prod87`.

File: scripts/releases.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        "--release-date", default=None, help="Date of the release")
+    parser.add_argument(
+        "--db-size", default=None, help="Size of the database for the release")
+    parser.add_argument("--format", default="tsv", help=(
+        "Output format, includes 'string', 'csv', 'tsv' (default), and "
+        "'json'. "
+        "'tsv' can be appended to the IDR studies.csv file with no further "
+        "processing. "
+        "All other formats include headers and totals. "
+        "'string' is the most human readable (fixed width columns). "
```
Comment (mtbc) [inline scripts/releases.py:83]:
```suggestion
        "'string' is the most human-readable (fixed width columns)."
```

**Comment by sbesson** [comment]
The last commit in the script fixes the empty line issue by passing a csv file to the `pandas.DataFrame.to_csv` method. It also introduces extra logic assuming that the script is run against `studies.tsv` in the idr.openmicroscopy.org repos by checking whether a `releases.tsv` file is present in the same directory and appending the new release stat to this file. To test it,

`venv/bin/python scripts/releases.py  /opt/IDR/idr.openmicroscopy.org/_data/studies.tsv --release-date 2020-09-03 --db-size 412.649968287`

should add a line to `releases.tsv` similar to https://github.com/IDR/idr.openmicroscopy.org/blob/88a32319675166cd0959a445d0eb6bfd31f95c61/_data/releases.tsv#L43

**Comment by pwalczysko** [comment]
Not changed in this PR, but the line 80 of the script, which contains the verbose help output, seems misleading. A ``tsv`` output can be appended to a ``studies.csv`` file... ? Should that not be ``releases.tsv`` instead ?
https://github.com/IDR/idr-utils/blob/daeb58de8be5965efa57055c898fcf18f4686de5/scripts/releases.py#L80

File: scripts/releases.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    parser.add_argument(
+        "--release", default=None, help="Name of the release")
+    parser.add_argument(
+        "--release-date", default=None, help="Date of the release")
+    parser.add_argument(
+        "--db-size", default=None, help="Size of the database for the release")
+    parser.add_argument("--format", default="tsv", help=(
+        "Output format, includes 'string', 'csv', 'tsv' (default), and "
+        "'json'. "
+        "'tsv' can be appended to the IDR studies.csv file with no further "
```
Comment (pwalczysko) [inline scripts/releases.py:80]: This line 80 of the script, which contains the verbose help output, seems misleading. A ``tsv`` output can be appended to a ``studies.csv`` file... ? Should that not be ``releases.tsv`` instead ?

File: scripts/releases.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+
+def main():
+    parser = argparse.ArgumentParser("Generate release statistics")
+    parser.add_argument(
+        "--release", default=None, help="Name of the release")
+    parser.add_argument(
+        "--release-date", default=None, help="Date of the release")
+    parser.add_argument(
+        "--db-size", default=None, help="Size of the database for the release")
```
Comment (pwalczysko) [inline scripts/releases.py:76]:
```suggestion
        "--db-size", default=None, help="Size of the database for the release in GB")
```

**Comment by pwalczysko** [comment]
Two suggestions for the help text, otherwise works as described, the empty lines are not produced anymore and the simplification of the script commands (no necessity to specify a path to ``releases.tsv`` is making things easier, thank you.

**Comment by sbesson** [comment]
Thanks @pwalczysko. Pushed an extra commit to clarify the description of the output and include the new workflow. Once this is signed off, the only remaining step might be to update the Submission workflow accordingly.

@dominikl I assume I should be able to close https://github.com/IDR/idr.openmicroscopy.org/issues/92 with a link to the various PRs or do you know of any remaining to do?

File: scripts/releases.py [hunk 4]
```diff
@@ -73,14 +73,17 @@ def main():
...
         "Output format, includes 'string', 'csv', 'tsv' (default), and "
         "'json'. "
-        "'tsv' can be appended to the IDR studies.csv file with no further "
-        "processing. "
         "All other formats include headers and totals. "
         "'string' is the most human-readable (fixed width columns)."
+        "If tsv is selected and a file called releases.tsv exists in the "
+        "same directory as the file specified by studies_file, the output "
+        "will be appended to this file."
+        "'string' is the most human-readable (fixed width columns)."
```
Comment (pwalczysko) [inline scripts/releases.py:86]:
```suggestion
```

This seems to be a repetition of https://github.com/IDR/idr-utils/pull/19/commits/eebda2c24183e573668da9468ab391a0e90f0418#diff-a20aed2e870f7cfe8de9cd8f4c2fa0c6R82 above, so suggesting a removal.

File: scripts/releases.py [hunk 5]
```diff
@@ -73,14 +73,17 @@ def main():
...
-        "--db-size", default=None, help="Size of the database for the release in GB")
+        "--db-size", default=None,
+        help="Size of the database for the release in GB")
     parser.add_argument("--format", default="tsv", help=(
         "Output format, includes 'string', 'csv', 'tsv' (default), and "
         "'json'. "
-        "'tsv' can be appended to the IDR studies.csv file with no further "
-        "processing. "
         "All other formats include headers and totals. "
         "'string' is the most human-readable (fixed width columns)."
```
Comment (pwalczysko) [inline scripts/releases.py:82]:
```suggestion
        "'string' is the most human-readable (fixed width columns). "
```

Comment (pwalczysko) [reply scripts/releases.py:82]: space is necessary to have the correct spacing for the next sentence
Screen Shot 2020-09-04 at 15 20 54

File: scripts/releases.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        "--release", default=None, help="Name of the release")
+    parser.add_argument(
+        "--release-date", default=None, help="Date of the release")
+    parser.add_argument(
+        "--db-size", default=None,
+        help="Size of the database for the release in GB")
+    parser.add_argument("--format", default="tsv", help=(
+        "Output format, includes 'string', 'csv', 'tsv' (default), and "
+        "'json'. "
+        "All other formats include headers and totals. "
```
Comment (pwalczysko) [inline scripts/releases.py:81]: Sorry, maybe one more suggestion. Is this sentence maybe saying  "All other formats except tsv ...." ? In that case, it should be moved to behind the text below, after "...will be appended to this file." ?

Comment (sbesson) [reply scripts/releases.py:81]: I am not even sure it's true. I'll remove. Effectively, we're primarily using `tsv`. This help text has been literally copied from `stats.py` initially.

Comment (sbesson) [reply scripts/releases.py]: a9c3a5c

**Review by pwalczysko — APPROVED** [review]

**Pull request merged**